src/Message/RedirectPurchaseRequest.php

```diff
@@ -173,4 +179,17 @@ public function sendData($data) : ResponseInterface
 
         return new RedirectPurchaseResponse($this, $data);
     }
+
+    /**
+     * @param CreditCard $card
+     *
+     * @return string
+     * @throws NumberParseException
+     */
+    private function formattedPhoneNumber(CreditCard $card): string
+    {
+        $phoneUtil = PhoneNumberUtil::getInstance();
+        $numberProto = $phoneUtil->parse($card->getPhone(), $this->getCountry($card->getBillingCountry(), 'alpha2'));
+        return $numberProto->getCountryCode()."|".$card->getPhone();
```

Comment:
I don't think this is right. This driver needs to strip any leading zeroes and remove any spaces from the phone number.

For example, I tried the phone number "01234 112233" and this method returned "44|01234 112233", but it should have returned "44|1234112233".

I'm guessing you can use `$numberProto->getNationalNumber()` instead of `$card->getPhone()` in the return expression.

Reply:
Can't use `getNationalNumber` as it only uses `trim` which only removes while spaces from the start and end. I'll use `str_replace`

Reply:
I'm not sure if this is right still as it might need to also remove the leading 0 from the number, but the documentation really isn't clear.

The example in the documentation shows them posting a value of "44|789456123", but it doesn't say whether the original number was "789 456123" or "0789 456123". I hate payment gateway documentation!!!

I notice the REGEX for that part of the value is `[0-9]{1,15}`, so it accepts leading zeroes. It also doesn't complain when I make the payment. I think we should leave it as it is and see if there are any problems after we release it.